hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsOutputStream.java

```diff
@@ -376,27 +384,31 @@ private synchronized void writeCurrentBufferToService() throws IOException {
     position += bytesLength;
 
     if (threadExecutor.getQueue().size() >= maxRequestsThatCanBeQueued) {
-      long start = System.currentTimeMillis();
-      waitForTaskToComplete();
-      outputStreamStatistics.timeSpentTaskWait(start, System.currentTimeMillis());
+      //Tracking time spent on waiting for task to complete.
+      try (DurationTracker ignored = outputStreamStatistics.timeSpentTaskWait()) {
+        waitForTaskToComplete();
+      }
     }
 
-    final Future<Void> job = completionService.submit(new Callable<Void>() {
-      @Override
-      public Void call() throws Exception {
-        AbfsPerfTracker tracker = client.getAbfsPerfTracker();
-        try (AbfsPerfInfo perfInfo = new AbfsPerfInfo(tracker,
-                "writeCurrentBufferToService", "append")) {
-          AbfsRestOperation op = client.append(path, offset, bytes, 0,
-                  bytesLength, cachedSasToken.get(), false);
-          cachedSasToken.update(op.getSasToken());
-          perfInfo.registerResult(op.getResult());
-          byteBufferPool.putBuffer(ByteBuffer.wrap(bytes));
-          perfInfo.registerSuccess(true);
-          return null;
-        }
-      }
-    });
+    final Future<Void> job =
+        completionService.submit(IOStatisticsBinding
+            .trackDurationOfCallable((IOStatisticsStore) ioStatistics,
+                AbfsStatistic.TIME_SPENT_ON_PUT_REQUEST.getStatName(),
+                () -> {
+                  AbfsPerfTracker tracker = client.getAbfsPerfTracker();
```

Comment:
given we are wrapping callables with callables, maybe the Abfs Perf tracker could join in. Not needed for this patch, but later...